File: 101-109-smallproblems/easy1/howmany.rb

def count_occurrences(array)
  occurrences = {}

  array.each do |element|
    # if occurrences.key? element
    #   occurrences[element] += 1
    # else
    #   occurrences[element] = 1
    # end
    occurrences[element] = array.count(element)
  end

  occurrences.each { |element, count| puts "#{element} => #{count}" }
end

## Examples

vehicles = ['car', 'car', 'truck', 'car', 'SUV', 'truck', 'motorcycle', 'motorcycle', 'car', 'truck']
count_occurrences(vehicles)

File: 210/small-problems/easy4/how-many.js
function countOccurrences(array) {
  var occurrences = {};

  array.forEach(function (a) {
    occurrences[a] = (occurrences[a] || 0) + 1;
  });

  Object.keys(occurrences).forEach(function (key) {
    console.log(key + ' => ' + String(occurrences[key]));
  });
}

// Example

var vehicles = ['car', 'car', 'truck', 'car', 'SUV', 'truck',
                'motorcycle', 'motorcycle', 'car', 'truck'];

countOccurrences(vehicles);

// car => 4
// truck => 3
// SUV => 1
// motorcycle => 2
